### ISSUE TITLE
Can't Install on Beta14

### ISSUE THREAD

**Issue by XiaoPika** [FriendsOfFlarum/upload#244]
```
Flarum core 0.1.0-beta.14.1
PHP version: 7.3.24
Loaded extensions: Core, date, libxml, openssl, pcre, sqlite3, zlib, bcmath, ctype, curl, dom, filter, ftp, gd, gettext, hash, iconv, intl, json, mbstring, SPL, session, pcntl, standard, PDO, mysqlnd, pdo_sqlite, bz2, posix, Reflection, pdo_mysql, shmop, SimpleXML, soap, sockets, sodium, mysqli, sysvsem, tokenizer, xml, xmlreader, xmlrpc, xmlwriter, zip, fileinfo, exif, yac, Phar, sysvshm, sysvmsg, wddx, gmp, imagick, redis, Zend OPcache
+--------------------------------------+------------------+--------+
| Flarum Extensions                    |                  |        |
+--------------------------------------+------------------+--------+
| ID                                   | Version          | Commit |
+--------------------------------------+------------------+--------+
| flarum-approval                      | v0.1.0-beta.14   |        |
| flarum-bbcode                        | v0.1.0-beta.12   |        |
| flarum-flags                         | v0.1.0-beta.14.1 |        |
| flarum-likes                         | v0.1.0-beta.14   |        |
| flarum-lock                          | v0.1.0-beta.14   |        |
| flarum-markdown                      | v0.1.0-beta.14   |        |
| flarum-mentions                      | v0.1.0-beta.14   |        |
| flarum-sticky                        | v0.1.0-beta.14   |        |
| flarum-subscriptions                 | v0.1.0-beta.14   |        |
| flarum-suspend                       | v0.1.0-beta.14   |        |
| flarum-tags                          | v0.1.0-beta.14   |        |
| michaelbelgium-discussion-views      | v5.0.0           |        |
| squeevee-fancybox                    | 0.2-beta         |        |
| fof-pages                            | 0.5.0            |        |
| fof-links                            | 0.4.0            |        |
| littlegolden-lang-simplified-chinese | v0.1.68          |        |
| nearata-remove-pane                  | v1.1.0           |        |
| nearata-copy-code-to-clipboard       | v1.1.0           |        |
| fof-formatting                       | 0.2.0            |        |
| fof-profile-image-crop               | 0.2.0            |        |
| fof-username-request                 | 0.3.0            |        |
| clarkwinkelmann-first-post-approval  | 0.1.2            |        |
| itnt-backtop                         | v0.1.7.2         |        |
| fof-best-answer                      | 0.2.0            |        |
| fof-filter                           | 0.1.0-beta.3     |        |
| fof-user-bio                         | 0.3.1            |        |
+--------------------------------------+------------------+--------+
Base URL: ***
Installation path: /www/wwwroot/flarum
Debug mode: off

[root@Pika-Web flarum]# composer require fof/upload
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Continue as root/super user [yes]? y
Using version ^0.10.0 for fof/upload
./composer.json has been updated
Running composer update fof/upload
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires fof/upload ^0.10.0 -> satisfiable by fof/upload[0.10.0].
    - fof/upload 0.10.0 requires flarum/core >=0.1.0-beta.12 <0.1.0-beta.14 -> found flarum/core[v0.1.0-beta.12, v0.1.0-beta.13] but the package is fixed to v0.1.0-beta.14.1 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

```

Thank you for all of your contributing!

**Comment by XiaoPika** [comment]
by the way, Beta14 defalut extension `Statistics  :  Add a basic statistics widget on the Dashboard.` seem not be compatible with beta14

**Comment by XiaoPika** [comment]
Oh i found the problem: the compatible version of `fof/upload` is `v0.11.0`, and there was` v0.10.0` by using the default install command.

**Comment by luceos** [comment]
Was this fixed then @XiaoPika ?

**Comment by XiaoPika** [comment]
unfortunately, no……

![](https://user-images.githubusercontent.com/10910848/98938756-5a0fcc80-2523-11eb-8e1f-1ef0c0538e89.jpg)

**Comment by XiaoPika** [comment]
I use the composer mirro due to [some reason](https://en.wikipedia.org/wiki/Internet_censorship_in_China), is that a problem?
```
[root@Pika-Web flarum]# composer -vvv require fof/upload
Reading ./composer.json
Loading config file /root/.config/composer/config.json
Loading config file /root/.config/composer/auth.json
Loading config file ./composer.json
Checked CA file /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem: valid
Executing command (/www/wwwroot/flarum): git branch -a --no-color --no-abbrev -v
Executing command (/www/wwwroot/flarum): git describe --exact-match --tags
Executing command (CWD): git --version
Executing command (/www/wwwroot/flarum): git log --pretty="%H" -n1 HEAD
Executing command (/www/wwwroot/flarum): hg branch
Executing command (/www/wwwroot/flarum): fossil branch list
Executing command (/www/wwwroot/flarum): fossil tag list
Executing command (/www/wwwroot/flarum): svn info --xml
Failed to initialize global composer: Composer could not find the config file: /root/.config/composer/composer.json
To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ "Getting Started" section
Reading /www/wwwroot/flarum/vendor/composer/installed.json
Running 2.0.6 (2020-11-07 11:21:17) with PHP 7.3.24 on Linux / 3.10.0-1127.19.1.el7.x86_64
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Continue as root/super user [yes]? y
Downloading https://mirrors.aliyun.com/composer/packages.json
[200] https://mirrors.aliyun.com/composer/packages.json
Writing /root/.cache/composer/repo/https---mirrors.aliyun.com-composer/packages.json into cache
Reading /root/.cache/composer/repo/https---mirrors.aliyun.com-composer/provider-fof~upload.json from cache
Downloading https://mirrors.aliyun.com/composer/p2/fof/upload.json if modified
[304] https://mirrors.aliyun.com/composer/p2/fof/upload.json
Using version ^0.10.0 for fof/upload
./composer.json has been updated
Reading ./composer.json
Loading config file /root/.config/composer/config.json
Loading config file /root/.config/composer/auth.json
Loading config file ./composer.json
Executing command (/www/wwwroot/flarum): git branch -a --no-color --no-abbrev -v
Executing command (/www/wwwroot/flarum): git describe --exact-match --tags
Executing command (/www/wwwroot/flarum): git log --pretty="%H" -n1 HEAD
Executing command (/www/wwwroot/flarum): hg branch
Executing command (/www/wwwroot/flarum): fossil branch list
Executing command (/www/wwwroot/flarum): fossil tag list
Executing command (/www/wwwroot/flarum): svn info --xml
Failed to initialize global composer: Composer could not find the config file: /root/.config/composer/composer.json
To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ "Getting Started" section
Reading /www/wwwroot/flarum/vendor/composer/installed.json
Running composer update fof/upload
Reading ./composer.lock
Loading composer repositories with package information
Updating dependencies
Downloading https://mirrors.aliyun.com/composer/packages.json
[200] https://mirrors.aliyun.com/composer/packages.json
Writing /root/.cache/composer/repo/https---mirrors.aliyun.com-composer/packages.json into cache
Downloading https://mirrors.aliyun.com/composer/p2/psr/http-message-implementation.json
Reading /root/.cache/composer/repo/https---mirrors.aliyun.com-composer/provider-fof~upload.json from cache
Downloading https://mirrors.aliyun.com/composer/p2/fof/upload.json if modified
[304] https://mirrors.aliyun.com/composer/p2/fof/upload.json
[404] https://mirrors.aliyun.com/composer/p2/psr/http-message-implementation.json
Reading /root/.cache/composer/repo/https---mirrors.aliyun.com-composer/provider-ramsey~uuid.json from cache
Downloading https://mirrors.aliyun.com/composer/p2/ramsey/uuid.json if modified
Reading /root/.cache/composer/repo/https---mirrors.aliyun.com-composer/provider-softcreatr~php-mime-detector.json from cache
Downloading https://mirrors.aliyun.com/composer/p2/softcreatr/php-mime-detector.json if modified
Reading /root/.cache/composer/repo/https---mirrors.aliyun.com-composer/provider-guzzlehttp~guzzle.json from cache
Downloading https://mirrors.aliyun.com/composer/p2/guzzlehttp/guzzle.json if modified
[304] https://mirrors.aliyun.com/composer/p2/softcreatr/php-mime-detector.json
[304] https://mirrors.aliyun.com/composer/p2/ramsey/uuid.json
[304] https://mirrors.aliyun.com/composer/p2/guzzlehttp/guzzle.json
Reading /root/.cache/composer/repo/https---mirrors.aliyun.com-composer/provider-paragonie~random-compat.json from cache
Downloading https://mirrors.aliyun.com/composer/p2/paragonie/random_compat.json if modified
Reading /root/.cache/composer/repo/https---mirrors.aliyun.com-composer/provider-guzzlehttp~promises.json from cache
Downloading https://mirrors.aliyun.com/composer/p2/guzzlehttp/promises.json if modified
[304] https://mirrors.aliyun.com/composer/p2/guzzlehttp/promises.json
[304] https://mirrors.aliyun.com/composer/p2/paragonie/random_compat.json
Generating rules
Resolving dependencies through SAT
Looking at all rules.

Dependency resolution completed in 0.001 seconds
Reading /root/.cache/composer/repo/https---mirrors.aliyun.com-composer/provider-flarum~core.json from cache
Downloading https://mirrors.aliyun.com/composer/p2/flarum/core.json if modified
[304] https://mirrors.aliyun.com/composer/p2/flarum/core.json
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires fof/upload ^0.10.0 -> satisfiable by fof/upload[0.10.0].
    - fof/upload 0.10.0 requires flarum/core >=0.1.0-beta.12 <0.1.0-beta.14 -> found flarum/core[v0.1.0-beta.12, v0.1.0-beta.13] but the package is fixed to v0.1.0-beta.14.1 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.
[root@Pika-Web flarum]# 
```

**Comment by luceos** [comment]
Try `composer update fof/upload -W --no-dev -a`

**Comment by XiaoPika** [comment]
```
[root@Pika-Web ~]# cd /www/wwwroot/flarum
[root@Pika-Web flarum]# composer update fof/upload -W --no-dev -a
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Continue as root/super user [yes]? y
Loading composer repositories with package information
Updating dependencies
Package "fof/upload" listed for update is not locked.
Nothing to modify in lock file
Installing dependencies from lock file
Nothing to install, update or remove
Generating optimized autoload files
65 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
```

**Comment by XiaoPika** [comment]
I cancel the mirror source of Composer and it's done, thank you honored  @luceos 
```
[root@Pika-Web flarum]# composer require fof/upload
Using version ^0.10.0 for fof/upload
./composer.json has been updated
Running composer update fof/upload
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires fof/upload ^0.10.0 -> satisfiable by fof/upload[0.10.0].
    - fof/upload 0.10.0 requires flarum/core >=0.1.0-beta.12 <0.1.0-beta.14 -> found flarum/core[v0.1.0-beta.12, v0.1.0-beta.13] but the package is fixed to v0.1.0-beta.14.1 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

[root@Pika-Web flarum]# composer config -g --unset repos.packagist

Do not run Composer as root/super user! See https://getcomposer.org/root for details

[root@Pika-Web flarum]# composer require fof/upload

Do not run Composer as root/super user! See https://getcomposer.org/root for details
Continue as root/super user [yes]? y
Using version ^0.11.0 for fof/upload
./composer.json has been updated
Running composer update fof/upload
Loading composer repositories with package information
Updating dependencies
Lock file operations: 6 installs, 0 updates, 0 removals
  - Locking fof/upload (0.11)
  - Locking guzzlehttp/guzzle (6.5.5)
  - Locking guzzlehttp/promises (1.4.0)
  - Locking paragonie/random_compat (v9.99.99)
  - Locking ramsey/uuid (3.9.3)
  - Locking softcreatr/php-mime-detector (3.0.0)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 6 installs, 0 updates, 0 removals
  - Downloading softcreatr/php-mime-detector (3.0.0)
  - Downloading paragonie/random_compat (v9.99.99)
  - Downloading ramsey/uuid (3.9.3)
  - Downloading guzzlehttp/promises (1.4.0)
  - Downloading guzzlehttp/guzzle (6.5.5)
  - Downloading fof/upload (0.11)
  - Installing softcreatr/php-mime-detector (3.0.0): Extracting archive
  - Installing paragonie/random_compat (v9.99.99): Extracting archive
  - Installing ramsey/uuid (3.9.3): Extracting archive
  - Installing guzzlehttp/promises (1.4.0): Extracting archive
  - Installing guzzlehttp/guzzle (6.5.5): Extracting archive
  - Installing fof/upload (0.11): Extracting archive
9 package suggestions were added by new dependencies, use `composer suggest` to see details.
Generating autoload files
66 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
```
